src/Common/Core/Impl/OS/Win32MessageBox.cs

```diff
@@ -72,8 +73,48 @@ public enum Result : uint {
                 form.BringToFront();
                 // Finally show the MessageBox with the form just created as its owner
                 flags |= Flags.ApplicationModal | Flags.Topmost | Flags.SetForeground;
-                return (Result)NativeMethods.MessageBox(form.Handle, message, "Microsoft Visual Studio", (uint)flags);
+
+                // Find VS window rectangle
+                var vsWindowRect = new NativeMethods.RECT();
+                NativeMethods.GetWindowRect(appWindowHandle, out vsWindowRect);
+
+                // Position message box in a way so it appears over VS but above the progress window
+                var centerX = vsWindowRect.Left + vsWindowRect.Width / 2;
+                var centerY = vsWindowRect.Top + vsWindowRect.Height / 2;
+
+                MoveMessageBoxAsync(Resources.MessageBoxTitle, centerX, centerY).DoNotWait();
+                return (Result)NativeMethods.MessageBox(form.Handle, message, Resources.MessageBoxTitle, (uint)flags);
             }
         }
+
+        private static Task MoveMessageBoxAsync(string title, int centerX, int centerY) {
+            return Task.Run(() => {
+                IntPtr msgBoxHandle = IntPtr.Zero;
+                var startTime = DateTime.Now;
+
+                while (msgBoxHandle == IntPtr.Zero && (DateTime.Now - startTime).TotalMilliseconds < 2000) {
```

Comment:
It is trying to locate message box handle. This task starts BEFORE the message box is shown so it can'd find the handle right away.